### PR TITLE
IDENTITY-3506 : Listing Workflow Pending Entities 

### PR DESCRIPTION
**Pull request by kasungayan**

### CODE REVIEW

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 1]
```diff
@@ -328,25 +366,26 @@
...
                 <tbody>
                 <%
-                    if(roles != null && roles.length > 0) {
-                         for (FlaggedName data : roles) {
-                            if (data != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug in Axis!!
-                                if(CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(data.getItemName())) {
+                    if (activeRoleList != null) {
+                        for (Iterator<FlaggedName> iterator = activeRoleList.iterator(); iterator.hasNext(); ) {
+                            FlaggedName flaggedName = iterator.next();
+                            if (flaggedName != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug in Axis!!
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: this comment seems unnecessary

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 2]
```diff
@@ -385,7 +431,92 @@
...
                         }
                     }
-               %>
+                %>
+
+                <%
+                    if (showDeletePendingRoles != null) {
+                        for (Iterator<FlaggedName> iterator = showDeletePendingRoles.iterator(); iterator.hasNext(); ) {
+                            FlaggedName flaggedName = iterator.next();
+                            if (flaggedName != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug in Axis!!
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: this comment seems unnecessary

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 3]
```diff
@@ -385,7 +431,92 @@
...
+                                if (userRealmInfo.getAdminRole().equals(flaggedName.getItemName()) &&
+                                        !userRealmInfo.getAdminUser().equals(currentUser)) {
+                                    continue;
+                                }
+                                String roleName = CharacterEncoder.getSafeText(flaggedName.getItemName());
+                                String disPlayName = CharacterEncoder.getSafeText(flaggedName.getItemDisplayName());
+                                if (disPlayName == null) {
+                                    disPlayName = roleName;
+                                }
+                                String disPlayRoleName = disPlayName + " " + "[Pending Roles for Delete]";
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Concat the two Strings.
The suffix should be changed as "Pending to be deleted" ?

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 4]
```diff
@@ -385,7 +431,92 @@
...
+
+                <%
+                    if (workFlowAddPendingRoles != null) {
+                        for (Iterator<String> iterator = workFlowAddPendingRoles.iterator(); iterator.hasNext(); ) {
+                            String role = iterator.next();
+                            if (role != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug Axis!!
+                                if (role.equals(CarbonConstants.REGISTRY_ANONNYMOUS_USERNAME)) {
+                                    continue;
+                                }
+                                String disPlayName = role + " " + "[Pending Roles for Add]";
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Concat the two Strings.
The suffix should be changed as "Pending to be added" ?

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 5]
```diff
@@ -414,24 +545,24 @@
...
-            for(int i=0;i<domains.length;i++){
-                arg += "\'"+domains[i]+"\'";
-                if(i < domains.length - 2){
-                    arg += ", ";
-                }else if(i == domains.length - 2){
-                    arg += " and ";
+            %>
+            <strong><%=message%></strong>
+            <%
+            }else if(exceededDomains.getItemDisplayName() != null && !exceededDomains.getItemDisplayName().equals("")){
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Format the code
Use StringUtils.isNotBlank()

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp [hunk 6]
```diff
@@ -414,24 +545,24 @@
...
+                    }else if(i == domains.length - 2){
+                        arg += " and ";
+                    }
                 }
-            }
-            message = resourceBundle.getString("more.roles").replace("{0}",arg);
-        %>
-        <strong><%=message%></strong>
-        <%
+                message = resourceBundle.getString("more.roles").replace("{0}",arg);
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Use MessageFormat.format()

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 7]
```diff
@@ -91,12 +101,12 @@
...
         claimUri = (java.lang.String) session.getAttribute(UserAdminUIConstants.USER_CLAIM_FILTER);
     }
-    session.setAttribute(UserAdminUIConstants.USER_CLAIM_FILTER,claimUri);
+    session.setAttribute(UserAdminUIConstants.USER_CLAIM_FILTER, claimUri);
     exceededDomains = (FlaggedName) session.getAttribute(UserAdminUIConstants.USER_LIST_CACHE_EXCEEDED);
 
     //  search filter
     String selectedDomain = request.getParameter("domain");
-    if(selectedDomain == null || selectedDomain.trim().length() == 0){
+    if (selectedDomain == null || selectedDomain.trim().length() == 0) {
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Use StringUtils.isBlank()

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 8]
```diff
@@ -436,40 +481,149 @@
...
+                            }
+                        }
+                    }
+                %>
+
+                <%
+                    if (showDeletePendingUsers != null) {
+                        for (Iterator<FlaggedName> iterator = showDeletePendingUsers.iterator(); iterator.hasNext(); ) {
+                            FlaggedName flaggedName = iterator.next();
+                            if (flaggedName != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug Axis!!
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Remove the comment

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 9]
```diff
@@ -436,40 +481,149 @@
...
+                                if (flaggedName.getItemName().equals(CarbonConstants.REGISTRY_ANONNYMOUS_USERNAME)) {
+                                    continue;
+                                }
+                                String userName = CharacterEncoder.getSafeText(flaggedName.getItemName());
+                                String disPlayName = CharacterEncoder.getSafeText(flaggedName.getItemDisplayName()) +
+                                        " " + "[Pending Users for Delete]";
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Merge the two strings
Change to "Pending to be deleted" ?

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 10]
```diff
@@ -436,40 +481,149 @@
...
+                                if (disPlayName == null || disPlayName.trim().length() == 0) {
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: use StringUtils.isNotBlank()

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 11]
```diff
@@ -481,6 +635,30 @@
...
+
+                <%
+                    if (workFlowAddPendingUsers != null) {
+                        for (Iterator<String> iterator = workFlowAddPendingUsers.iterator(); iterator.hasNext(); ) {
+                            String user = iterator.next();
+                            if (user != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug Axis!!
+                                if (user.equals(CarbonConstants.REGISTRY_ANONNYMOUS_USERNAME)) {
+                                    continue;
+                                }
+                                String disPlayName = user + " " + "[Pending Users for Add]";
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Merge the two strings
Change to "Pending to be added" ?

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 12]
```diff
@@ -491,45 +669,47 @@
...
 
             <%
-                if (users != null && users.length > 0 && exceededDomains != null) {
-                    if(exceededDomains.getItemName() != null || exceededDomains.getItemDisplayName() != null){
+                if (activeUserList.size() > 0 && exceededDomains != null) {
+                    if (exceededDomains.getItemName() != null || exceededDomains.getItemDisplayName() != null) {
                         String message = null;
-                        if(exceededDomains.getItemName() != null && exceededDomains.getItemName().equals("true")){
-                            if(exceededDomains.getItemDisplayName() != null && !exceededDomains.getItemDisplayName().equals("")){
+                        if (exceededDomains.getItemName() != null && exceededDomains.getItemName().equals("true")) {
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Change to .....&& Boolean.valueOf(exceededDomains.getItemName())

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 13]
```diff
@@ -491,45 +669,47 @@
...
+                            if (exceededDomains.getItemDisplayName() != null && !exceededDomains.getItemDisplayName().equals("")) {
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Use StringUtils

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 14]
```diff
@@ -491,45 +669,47 @@
...
+                                    if (i < domains.length - 2) {
                                         arg += ", ";
-                                    }else if(i == domains.length - 2){
+                                    } else if (i == domains.length - 2) {
                                         arg += " and ";
                                     }
                                 }
-                                message = resourceBundle.getString("more.users.others").replace("{0}",arg);
-                            } else{
+                                message = resourceBundle.getString("more.users.others").replace("{0}", arg);
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: use MessageFormatter.format

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 15]
```diff
@@ -491,45 +669,47 @@
...
-                            }
-                            message = resourceBundle.getString("more.users").replace("{0}",arg);
-        %>
-                <strong><%=message%></strong>
-        <%
+            %>
+            <strong><%=message%>
+            </strong>
+            <%
+            } else if (exceededDomains.getItemDisplayName() != null && !exceededDomains.getItemDisplayName().equals("")) {
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: use StringUtils

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp [hunk 16]
```diff
@@ -491,45 +669,47 @@
...
+                String arg = "";
+                for (int i = 0; i < domains.length; i++) {
+                    arg += "\'" + domains[i] + "\'";
+                    if (i < domains.length - 2) {
+                        arg += ", ";
+                    } else if (i == domains.length - 2) {
+                        arg += " and ";
+                    }
+                }
+                message = resourceBundle.getString("more.users").replace("{0}", arg);
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Use MessageFormatter.format()

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow.ui/pom.xml [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>org.wso2.carbon.user.mgt.workflow.ui.*</Export-Package>
+                        <Import-Package>
+                            org.wso2.carbon.ui;version="${carbon.kernel.package.import.version.range}",
+                            org.apache.axis2,
+                            org.apache.axis2.client,
+                            org.apache.axis2.context,
+                            org.apache.commons.logging,
+                            org.wso2.carbon.user.mgt.workflow.stub
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow.ui/pom.xml]: Add import version ranges

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 1 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 2 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 3 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #889

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 4 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #889

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 5 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp (hunk 6 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 7 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 8 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 9 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #889

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 10 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 11 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 12 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 13 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 14 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 15 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #915

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp (hunk 16 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/user-mgt.jsp]: Fixed with #1248

File: components/user-mgt/org.wso2.carbon.user.mgt.workflow.ui/pom.xml (hunk 17 above)
Comment (kasungayan) [inline components/user-mgt/org.wso2.carbon.user.mgt.workflow.ui/pom.xml]: Fixed with #1248

**Pull request merged**